tket2/src/serialize/pytket/encoder.rs

```diff
@@ -619,8 +632,28 @@ impl ParameterTracker {
                 // Re-use the parameter from the input.
                 inputs[0].clone()
             }
-            OpType::ExtensionOp(_) if optype.cast() == Some(RotationOp::radd) => {
-                format!("{} + {}", inputs[0], inputs[1])
+            // Encode some angle and float operations directly as strings using
+            // the already encoded inputs. Fail if the operation is not
+            // supported, and let the operation encoding process it instead.
+            OpType::ExtensionOp(_) => {
+                if let Some(s) = optype
+                    .cast::<RotationOp>()
+                    .and_then(|op| self.encode_rotation_op(&op, inputs.as_slice()))
+                {
+                    s
+                } else if let Some(s) = optype
+                    .cast::<FloatOps>()
+                    .and_then(|op| self.encode_float_op(&op, inputs.as_slice()))
+                {
+                    s
+                } else if let Some(s) = optype
+                    .cast::<SympyOp>()
+                    .and_then(|op| self.encode_sympy_op(&op, inputs.as_slice()))
+                {
+                    s
+                } else {
+                    return Ok(false);
+                }
```

Comment:
Any way this could be some sort of `match` statement on the optype type instead of trying each out?

Reply:
Each cast has a different type, so all options I tried for writing this look messy.